### PR TITLE
RANDAO hash chain and no stale shuffles

### PR DESCRIPTION
**Pull request by JustinDrake**
* Make RANDAO into a hash chain (this makes it easy for applications to prove the intermediate RANDAO reveals).
* Include `state.slot` when shuffle to avoid stale shuffles with skip slots

### CODE REVIEW

**Comment by vbuterin** [comment]
Looks fine to me!

Though keep in mind that `get_new_shuffling` is being renamed to `get_shuffling` in the other PR.

**Comment by djrtwo** [comment]
handling merge conflict as  I review @JustinDrake

File: specs/core/0_beacon-chain.md [hunk 1]
```diff
@@ -880,7 +880,7 @@ def get_shuffling(seed: Hash32,
     )
 
     # Shuffle
-    seed = xor(randao_mix, Hash32(slot))
+    seed = xor(randao_mix, hash(slot))
```
Comment (JustinDrake) [inline specs/core/0_beacon-chain.md]: The intent here was a type cast, not a call to the `hash` function.

Comment (djrtwo) [reply specs/core/0_beacon-chain.md]: ah

**Pull request merged**